llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -3839,3 +3839,14 @@ SIRegisterInfo::getSubRegAlignmentNumBits(const TargetRegisterClass *RC,
   }
   return 0;
 }
+
+SmallVector<SmallString<8>>
+SIRegisterInfo::getVRegFlagsOfReg(Register Reg,
+                                  const MachineFunction &MF) const {
+  SmallVector<SmallString<8>> RegFlags;
+  const SIMachineFunctionInfo *FuncInfo = MF.getInfo<SIMachineFunctionInfo>();
+  if (FuncInfo->checkFlag(Reg, AMDGPU::VirtRegFlag::WWM_REG)) {
+    RegFlags.push_back(SmallString<8>("WWM_REG"));
+  }
```

Comment:
```suggestion
  if (FuncInfo->checkFlag(Reg, AMDGPU::VirtRegFlag::WWM_REG))
    RegFlags.push_back("WWM_REG");
```

I would hope construct from literal works fine

Reply:
Should I change this to a const char* instead?

Reply:
I expect only literals, so could use StringRef or StringLiteral

Reply:
Alright, will keep this.
I have to keep the explicit construct since "WWM_REG" is being refused to be casted to a SmallString (without creating a StringRef explicitly)

Reply:
I mean the interface could change from SmallString to StringLiteral

Reply:
Got it